apps/backend/app/api/routes/data_issues.py

```
@@ -0,0 +1,696 @@
+    """
+    start_time = time.time()
+
+    try:
+        # Get the dataset
+        user_data = await UserData.find_one({
+            "user_id": current_user_id,
+            "_id": request.dataset_id
+        })
+
+        if not user_data:
+            raise HTTPException(status_code=404, detail="Dataset not found")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)